File: emulinker/src/main/java/org/emulinker/kaillera/model/event/UserQuitEvent.java
package org.emulinker.kaillera.model.event;

import org.emulinker.kaillera.model.*;

public class UserQuitEvent implements ServerEvent {
  private KailleraServer server;
  private KailleraUser user;
  private String message;

  public UserQuitEvent(KailleraServer server, KailleraUser user, String message) {
    this.server = server;
    this.user = user;
    this.message = message;
  }

  @Override
  public String toString() {
    return "UserQuitEvent";
  }

  @Override
  public KailleraServer getServer() {
    return server;
  }

  public KailleraUser getUser() {
    return user;
  }

  public String getMessage() {
    return message;
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/event/UserQuitEvent.kt
package org.emulinker.kaillera.model.event

import org.emulinker.kaillera.model.KailleraServer
import org.emulinker.kaillera.model.KailleraUser

class UserQuitEvent(
    override val server: KailleraServer, val user: KailleraUser, val message: String
) : ServerEvent {
  override fun toString(): String {
    return "UserQuitEvent"
  }
}
